Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs

```diff
@@ -50,6 +50,8 @@ public SpaceMode DistanceSpaceMode
                 // Convert world to local distances and vice versa whenever we switch the mode
                 if (value != distanceSpaceMode)
                 {
+                    // NOTE: This code is copied/pasted in PressableButtonInspector because
+                    // the inspector can only modify the serialized field and not the property
```

Comment:
can't we move this code into a public function called ConvertPlaneDistances(SpaceMode toMode) and then call this function here and in the inspector?

Reply:
if you do that you don't have to expose WorldToLocalScale and LocalToWorldScale public

Reply:
Yes, I chose not to surface the public method. We could make one. I guess it's really pick your poison. They're both ugly cases to handle this one scenario just for the editor.

Reply:
i'm observing a very strange behavior - I debugged a bit in the inspector because I thought I had used that setter before and seems like you can call the setter from the inspector code - however, for some reason the hololens2 version of the button doesn' t apply the changes properly. So PressableButtons work with calling the setter but Hololens 2 won't - They both however call that setter, so there must be something else interfering afterwards in the hololens 2 version of that button

Reply:
What about doing something like this?

                var pressableButton = (PressableButton)serializedObject.targetObject;
                pressableButton.DistanceSpaceMode = (PressableButton.SpaceMode)distanceSpaceMode.enumValueIndex;

Then the setter logic can stay in one location.  The thing I'm unsure of is how to make sure that the values don't get overwritten by a later call to serializedObject.ApplyModifiedProperties.  Is that what serializedObject.Update() does?  Copy the values from the real object onto the serialized representation?

Reply:
@AdamMitchell-ms the setter was being used in the original version. The Update() is supposed to copy the object values into the serializedobject and is why I had that originally. 

But exactly as the purpose for this PR being created, I had a lot of weird behavior using ApplyModifiedProperties & Update(). I tried multiple different iterations and values would just not get saved properly. 

The EditorGUILayout.PropertyField() is the only thing thus far that seems to cover all the bases correctly for whatever reason.

Reply:
it's only in the derived class where this is causing issues for me ¯\_(ツ)_/¯